### PR TITLE
feat: add foundational a11y documentation

### PR DESCRIPTION
**Pull request by josedialpad**
Jira ticket: https://switchcomm.atlassian.net/browse/DT-162

## Description
Added foundational a11y documentation to site.
New pages:
- Overview.
- Best Practices.
- Tooling.

## Pull Request Checklist

 - [x] Ask the contributors if a similar effort is already in process or has been solved.
 - [x] Review the [contribution guidelines](https://github.com/dialpad/dialtone/blob/staging/.github/CONTRIBUTING.md).
 - [x] Use `staging` as your pull request's base branch. (All PRs using `production` as its base branch will be declined).
 - [x] Ensure all `gulp` scripts successfully compile.
 - [x] Update, remove, or extend all affected documentation.
 - [ ] Ensure no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
 - [x] Request a review from Brad Paugh, David Becher, or Drew Chandler.

## Obligatory GIF (super important!)
gif

### CODE REVIEW

File: docs/_data/site-nav.json [hunk 1]
```diff
@@ -550,6 +550,27 @@
...
+          "title": "Best Practices",
+          "url": "/accessibility/best-practices/",
+          "publish": "yes"
+        },
+        {
+          "title": "Tooling",
+          "url": "/accessibility/tooling/",
+          "publish": "yes"
+        }
+      ]
```
Comment (dialpad-drew) [inline docs/_data/site-nav.json]: Can we avoid this as a top-level nav item and instead nest it under "Getting Started?"
Screen Shot 2021-11-29 at 7 30 27 PM

File: docs/accessibility/overview.html [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  <li>Help, such as contextual help and error prevention</li>
+  {% endul %}
+  {% header "h3", "Robust" %}
+  {% paragraph %}The interface and its content is designed and implemented in such a way that it works across a variety of users agents and assistive technologies.{% endparagraph %}
+  {% paragraph %}<strong>Examples:</strong>{% endparagraph %}
+  {% ul %}
+  <li>Parsing and generally ‘clean’ code like proper heading hierarchy, no duplicate attributes, complete start and end tags for elements</li>
+  <li>Name, role and value</li>
+  <li>Status messages, sending messaging to the user without interrupting their workflow by giving focus to an element. e.g. “added to cart” while shopping online should act as a message and not a redirection of activity and focus for the user</li>
+  {% endul %}
```
Comment (dialpad-drew) [inline docs/accessibility/overview.html]: Can we "chunk up" the POUR sections a little better with better spacing? It's kinda hard to tell them apart.
Screen Shot 2021-11-29 at 7 26 55 PM

File: docs/accessibility/tooling.html [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  {% paragraph %}<a href="https://webaim.org/articles/nvda/" class="d-link" target="_blank">WebAim - Using NVDA to Evaluate Web Accessibility</a>{% endparagraph %}
+  {% header "h3", "JAWS for Windows" %}
+  {% paragraph %}This software is not free and costs from $95/year for home use.{% endparagraph %}
+  {% paragraph %}<a href="https://www.freedomscientific.com/products/software/jaws/" class="d-link" target="_blank">Freedom Scientific - JAWS</a>{% endparagraph %}
+  {% paragraph %}<a href="https://webaim.org/articles/jaws/" class="d-link" target="_blank">WebAim - Using JAWS to Evaluate Web Accessibility</a>{% endparagraph %}
+  {% header "h3", "VoiceOver on iOS" %}
+  {% paragraph %}<a href="https://support.apple.com/en-us/HT210076" class="d-link" target="_blank">Apple - Turn on and practice VoiceOver on iPhone</a>{% endparagraph %}
+  {% header "h3", "Talkback for Android" %}
+  {% paragraph %}<a href="https://support.google.com/accessibility/android/answer/6007100?hl%3Den" class="d-link" target="_blank">Google - Turn TalkBack on or off</a>{% endparagraph %}
+</section>
```
Comment (dialpad-drew) [inline docs/accessibility/tooling.html]: I understand this is he we handle `h2` and `h3` tags, visually this page is a lot, both the table of contents as well as all the deep link icons. Anything we can do about that?
Screen Shot 2021-11-29 at 7 27 26 PM

Something more like this?
Screen Shot 2021-11-29 at 7 28 38 PM

File: docs/accessibility/overview.html [hunk 4]
```diff
@@ -0,0 +1,94 @@
+---
+layout: page
+title: Overview
+---
+<section class="d-stack16">
+  {% paragraph %}We are committed to experiences and products that are usable by everyone. Because “everyone” is a lot of people, reflecting a variety of backgrounds, abilities, and circumstances, we believe in applying thoughtful and inclusive design and development practices.{% endparagraph %}
+  {% paragraph %}Dialpad strives to maintain <strong>WCAG 2.1 AA</strong> compliance in our digital products.{% endparagraph %}
+  {% paragraph %}Everyone benefits when things are designed accessibly. Simply put: building products for accessibility is simply building products for better usability. Considering <strong>1 in 4</strong> US adults -- that’s 61 million people! -- live with a permanent disability and many more will have temporary or situational throughout their lifetime, accessibility isn’t for just some people. It benefits everyone.{% endparagraph %}
+  <div class="d-ta-center">
+    <img src="/assets/images/accessibility-graphic.png" alt="Accessibility graphic" />
```
Comment (dialpad-drew) [inline docs/accessibility/overview.html]: Can we have slightly more useful alt text here?

**Comment by dialpad-drew** [comment]
Hey dude, this is good! I added some feedback and made a couple super small changes in the Google Doc, which should be highlighted.

File: docs/_data/site-nav.json (hunk 1 above)
Comment (josedialpad) [reply docs/_data/site-nav.json]: Moved under the "Getting Started" section https://github.com/dialpad/dialtone/pull/512/commits/ae1260e44367b18b58c952cd2496ed723a06174b.

In the screenshot, "Fundamentals" is the first page for accessibility but we have "Overview" in the google doc. Which one is preferable?

File: docs/accessibility/overview.html (hunk 2 above)
Comment (josedialpad) [reply docs/accessibility/overview.html]: Updated the spacing of the principles https://github.com/dialpad/dialtone/pull/512/commits/c8dc9f0fd5fca39f3b69122459abec39584a2676

File: docs/accessibility/tooling.html (hunk 3 above)
Comment (josedialpad) [reply docs/accessibility/tooling.html]: Yeah, using raw `h3` without defined `id` we can skip the link and the TOC plugin https://github.com/dialpad/dialtone/pull/512/commits/1a1b40ab6df869a9934f62f0ee5485b9b19e7042

File: docs/accessibility/overview.html (hunk 4 above)
Comment (josedialpad) [reply docs/accessibility/overview.html]: Updated using the alt text from the source of the image https://github.com/dialpad/dialtone/pull/512/commits/3e080652509866349b69db836e29a195242f2d4d

**Comment by josedialpad** [comment]
> Hey dude, this is good! I added some feedback and made a couple super small changes in the Google Doc, which should be highlighted.

thanks for reviewing, I've updated the PR

File: docs/_data/site-nav.json (hunk 1 above)
Comment (dialpad-drew) [reply docs/_data/site-nav.json]: Actually, maybe "Fundamentals." Good call!

**Review by dialpad-drew — APPROVED** [review]
Approved with one small comment. Nice work, dude! 🙌

File: docs/_data/site-nav.json (hunk 1 above)
Comment (josedialpad) [reply docs/_data/site-nav.json]: Ok, changed page to "Fundamentals"  and added "Overview" as a header of the first section https://github.com/dialpad/dialtone/pull/512/commits/c2932c081b2c7f997639f8f7ea0a35a16046beec

**Pull request merged**